### PR TITLE
esConsumes migration for EDModules in SiStrip packages, part 2

### PR DESCRIPTION
**Pull request by pieterdavid**
#### PR description:

esConsumes migration https://github.com/cms-sw/cmssw/issues/31061 for modules in packages with SiStrip in the name not done in https://github.com/cms-sw/cmssw/pull/31826 (mostly DQM-related).
I tried to keep other changes minimal, but did some refactoring in SiStripMonitorCondData (making it use the helper class SiStripClasToMonitorCondData because they had a lot of duplicate code, and the esConsumes-related changes would have added even more); @sroychow @arossi83 could you have a look and comment on whether this is covered by the tests, or how to validate the changes otherwise?

#### PR validation:

compiles, `runTheMatrix.py -l limited --useInput all` passes and only shows differences in `TimerService`, but that may not cover all modified plugins.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@pieterdavid 

> could you have a look and comment on whether this is covered by the tests, or how to validate the changes otherwise?

it is used in the strips conddb monitoring webservice. It should be tested by using these scripts: https://twiki.cern.ch/twiki/bin/viewauth/CMS/StripTrackerMonitoringCondsDb

**Comment by pieterdavid** [comment]
@mmusich 
> it is used in the strips conddb monitoring webservice. It should be tested by using these scripts: https://twiki.cern.ch/twiki/bin/viewauth/CMS/StripTrackerMonitoringCondsDb

Thanks! I tried a few, and found some issues with the transitions for that plugins - they should be fixed with the last commit, as well as the failure in the header consistency check.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by yuanchao** [comment]
Hi, most of the EventSetupRecord::get function called are migrated to the suggested esConsumes. Also the WF matrix tests and unit tests are all passed. Other build warnings are not part of the modification. But still some of deprecated get(ESHandle<>) are not covered. Don't know if these can be fixed too? (runInfoRec->get(sumFED);)
/DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc 117
/DQM/SiStripMonitorClient/plugins/SiStripOfflineDQM.cc 118
/DQM/SiStripMonitorClient/plugins/SiStripCertificationInfo.cc 40
/DQM/SiStripMonitorSummary/plugins/SiStripCorrelateNoise.cc 99
/DQM/SiStripMonitorClient/plugins/SiStripDcsInfo.cc 57

**Comment by pieterdavid** [comment]
Indeed, it looks like I missed some cases... thanks for checking @yuanchao .

I have one doubt: most of these currently check for the presence of the `RunInfoRcd` as well as the validity of the `RunInfo` object in it, e.g. (from [here](https://github.com/cms-sw/cmssw/blob/master/DQM/SiStripMonitorClient/plugins/SiStripDaqInfo.cc#L119-L127)):
```C++
  auto runInfoRec = eSetup.tryToGet<RunInfoRcd>();
  if (!runInfoRec)
    return;
  edm::ESHandle<RunInfo> sumFED;
  runInfoRec->get(sumFED);
  if (!sumFED.isValid())
    return;
```
which I would migrate to
```C++
if (!eSetup.tryToGet<RunInfoRcd>)
  return;
auto sumFED = eSetup.getHandle(runInfoToken_);
if (!sumFED )
  return;
}
```
but I'm not sure if that wouldn't already fail at esConsumes in the constructor in case the `RunInfoRcd` isn't there (the [documentation](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module) does not mention an optional consume from ED modules).

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by yuanchao** [comment]
@pieterdavid Generally "RunInfoRcd" is needed for sub-systems to run. But sorry, not knowing about your use case so not be able to comment here.

+1
Thanks a lot for the update.

**Comment by yuanchao** [comment]
+1
Thanks a lot for the update.

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by makortel** [comment]
By quick look it seems that this PR could be causing a failure of `TestDQMOnlineClient-sistrip_dqm_sourceclient` unit test, see #32770

File: DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc [hunk 1]
```diff
@@ -71,17 +71,15 @@ void SiStripAnalyser::beginRun(edm::Run const& run, edm::EventSetup const& eSetu
...
-  unsigned long long const cacheID = eSetup.get<SiStripFedCablingRcd>().cacheIdentifier();
-  if (m_cacheID_ != cacheID) {
-    m_cacheID_ = cacheID;
+  if (fedCablingWatcher_.check(eSetup)) {
     edm::LogInfo("SiStripAnalyser") << "SiStripAnalyser::beginRun: "
                                     << " Change in Cabling, recrated TrackerMap";
-    if (!actionExecutor_.readTkMapConfiguration(eSetup)) {
+    detCabling_ = &eSetup.getData(detCablingToken_);
+    if (!actionExecutor_.readTkMapConfiguration(
+            detCabling_, &eSetup.getData(tkDetMapTokenER_), &eSetup.getData(tTopoTokenER_))) {
```
Comment (makortel) [inline DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc:79]: The token was declared to be consumed in `EndRun`, but is being used in `beginRun`.

File: DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc [hunk 2]
```diff
@@ -103,7 +101,8 @@ void SiStripAnalyser::analyze(edm::Event const& e, edm::EventSetup const& eSetup
       actionExecutor_.createDummyShiftReport();
     } else {
       auto& dqm_store = *edm::Service<DQMStore>{};
-      actionExecutor_.fillStatus(dqm_store, detCabling_, eSetup);
+      actionExecutor_.fillStatus(
+          dqm_store, detCabling_, &eSetup.getData(tkDetMapTokenELB_), &eSetup.getData(tTopoTokenELB_));
```
Comment (makortel) [inline DQM/SiStripMonitorClient/plugins/SiStripAnalyser.cc:105]: The token was declared to be consumed in `EndLuminosityBlock`, but is begin used in `analyze`.

**Comment by mmusich** [comment]
I gave a try to fix the issue #32770 at https://github.com/cms-sw/cmssw/pull/32771. 
Wondering why the DQM online client unit tests did not run in the PR tests here?
Can someone clarify how the unit tests to be run in a PR are chosen? Via dependencies?

**Comment by jfernan2** [comment]
I guess it is via dependencies yes, changes in DQM/Integration package in this case 
Not sure if experts want to modify the jenkins settings to run always for every PR unit tests which are targeting Online DQM world only
Thanks for the fix

**Comment by makortel** [comment]
> I gave a try to fix the issue #32770 at #32771.

Thanks!

> Can someone clarify how the unit tests to be run in a PR are chosen? Via dependencies?

Essentially the code of the PR is checked out, then `git cms-checkdeps -a` (i.e. all C++ and python source dependencies), and the unit tests of those packages are run. This misses unit tests in a package outside of the aforementioned set of packages that e.g. runs a configuration that loads a plugin defined in the aforementioned set of packages.

This is a known deficiency in the test system that has been discussed a few times. It is possible to ask the bot to run all unit tests, but IIUC the `full(_cmssw|)` would would lead to also building full CMSSW (@smuzaffar please clarify). So far such failures have been rare-enough for catching them in IBs to be sufficient (even if suboptimal).

**Comment by smuzaffar** [comment]
For now, one can explicilty ask to run tests for a package by setting the `addpkg = CMSSW/Package` test parameter. I am looking in to enabling full unit tests for each PR that should avoid such issues.

**Comment by makortel** [comment]
> For now, one can explicilty ask to run tests for a package by setting the `addpkg = CMSSW/Package` test parameter.

Thanks Shahzad. I remembered something was done, but didn't remember what exactly, and somehow missed the `addpkg` in http://cms-sw.github.io/cms-bot-cmssw-cmds.html.

**Comment by pieterdavid** [comment]
> I gave a try to fix the issue #32770 at #32771.

Thanks @mmusich !

**Comment by silviodonato** [comment]
see https://github.com/dmwm/T0/pull/4574

**Pull request merged**